azavea/pfb-network-connectivity
# Closes #273: Save job names/defs to AnalysisJob model

## Overview

The `settings.PFB_AWS_BATCH_ANALYSIS_JOB_DEFINITION_NAME_REVISION` and `settings.PFB_AWS_BATCH_TILEMAKER_JOB_DEFINITION_NAME_REVISION` values get incremented with every deployment, and we use this value to effectively track a few different things in the AnalysisJob model. By always lazily generating job names/definitions/etc via `@property` we lose the association to the actual values when a deployment occurs because revision is incremented. This PR changes the inner workings of the `job_name` properties so that we lazily save the proper value the first time these properties are accessed, and then just return that afterwards.


### Notes

Unable to use ModelField(default=callback) option
because we need the job uuid to create the job name, and args cannot be provided to the callback.

Other option was to make job name fields DB fields instead of
properties directly, but that requires additional save logic
and a data migration that would take more time to write,
and we don't filter/sort on these fields anyways. Could
still migrate to that later if necessary, as this puts the
first half of that machinery in place.

This solution feels a bit smelly and I'm not totally sold on it, in that we don't typically write Django models this way, and still has the disadvantage of being unable to filter/sort via ORM. But, the extra code required if we were to go for the solution in the paragraph above also feels meh (example in #275) because we have to duplicate code in data migrations. And it would take a bit longer to implement, using time which can be spent on other more pressing things ATM. 

## Testing Instructions

Run migrations. Create an analysis job via API, then access `job.analysis_job_name` and `job.tilemaker_job_name` via the Django shell. Should get the same job names as before. Access the `_` versions afterwards, and will see that the values are saved.

Closes #273
